### PR TITLE
build: enhance usage of libwhich path resolution to respect SONAME

### PR DESCRIPTION
**Pull request by vtjnash**
If the SONAME is available, it directly names the file we want to link to, so use that instead.

### CODE REVIEW

File: base/Makefile [hunk 1]
```diff
@@ -130,6 +130,14 @@ resolve_path += && \
 	$1_wd=`dirname "$${$1}"`; $1=$${$1_}; \
 	if [ -z "`echo $${$1} | grep '^/'`" ]; then $1=$${$1_wd}/$${$1}; fi; \
 	fi
+else
+# try to use the SO_NAME (if the named file exists)
+resolve_path += && \
+	$1_=`objdump -p "$${$1}" | awk '/SONAME/ {print $$2}'` && \
```
Comment (ararslan) [inline base/Makefile]: I guess Awk should be added to the README section on required build tools?

File: deps/llvm.mk [hunk 2]
```diff
@@ -58,6 +58,12 @@ LLVM_CPPFLAGS :=
 LLVM_LDFLAGS :=
 LLVM_CMAKE :=
 
+LLVM_CFLAGS :=
+LLVM_CXXFLAGS :=
+LLVM_CPPFLAGS :=
+LLVM_LDFLAGS :=
+LLVM_CMAKE :=
+
```
Comment (staticfloat) [inline deps/llvm.mk]: This looks to be an unrelated change

File: base/Makefile (hunk 1 above)
Comment (staticfloat) [reply base/Makefile]: I actually wonder if `awk` is not required by POSIX?  It's builtin to `busybox`, which is my poor-mans litmus test to know whether or not it's available everywhere.  ;)

**Review by staticfloat — CHANGES_REQUESTED** [review]

File: base/Makefile (hunk 1 above)
Comment (vtjnash) [reply base/Makefile]: fwiw, we already use it (to generate julia_version.h) – I checked before I used it here :)

**Comment by vtjnash** [comment]
@staticfloat updated

File: deps/llvm.mk (hunk 2 above)
Comment (staticfloat) [reply deps/llvm.mk]: @vtjnash is this actually important in this PR?

Comment (vtjnash) [reply deps/llvm.mk]: Not really ... and I just realized this is already in the Makefile right above it. Apparently this was just a rebase mistake – whereas I thought it was something I had done locally.
